price/src/price_generator.rs

```diff
@@ -233,36 +235,45 @@ impl PriceGenerator {
     }
 
     async fn get_pyth_price(&self, price_key: &SolPubkey) -> Result<Price> {
-        let mut account = self.client.get_account(price_key).await?;
-        let price_oracle = pyth_sdk_solana::load_price_feed_from_account(price_key, &mut account)?;
-        let curr_price = price_oracle
-            .get_ema_price_no_older_than(Utc::now().timestamp(), self.pyth_price_interval.as_secs())
-            .ok_or_else(|| anyhow!("No new price in the given interval"))?;
+        let account = self.client.get_account(price_key).await?;
+        let PriceUpdateV2 {
+            price_message:
+                PriceFeedMessage {
+                    ema_price: price,
+                    ema_conf: confidence,
+                    exponent,
+                    publish_time,
+                    ..
+                },
+            ..
```

Comment:
it's probably fewer lines to destructure this one level deep and then dot reference the fields: 
```suggestion
        let PriceUpdateV2 {
            price_message,
            ..
```